DiscordIan/Module/OpenWeather.cs

```diff
@@ -0,0 +1,317 @@
+            Discord.Embed embed;
+
+            if (coordinates == null || locale == null)
+                (message, embed) = await GetWeatherResultAsync(location);
+            else
+                (message, embed) = await GetWeatherResultAsync(coordinates, locale);
+
+
+            if (message == null)
+            {
+                await ReplyAsync("No weather found, sorry!");
+            }
+            else
+            {
+                await ReplyAsync(message, false, embed);
+            }
+        }
+
+        private async Task<(string, Discord.Embed)>
+            GetWeatherResultAsync(Model.Geocodio.Location coordinates, string location)
+        {
+            var headers = new Dictionary<string, string>
+            {
+                { "User-Agent", "DiscorIan Discord bot" }
+            };
+
+            var uriCurr = new Uri(string.Format(_options.IanOpenWeatherMapEndpointCoords,
+                HttpUtility.UrlEncode($"{coordinates.Lat}"),
+                HttpUtility.UrlEncode($"{coordinates.Lng}"),
+                _options.IanOpenWeatherKey));
+
+            var uriFore = new Uri(string.Format(_options.IanOpenWeatherMapEndpointForecast,
+                HttpUtility.UrlEncode($"{coordinates.Lat}"),
+                HttpUtility.UrlEncode($"{coordinates.Lng}"),
+                _options.IanOpenWeatherKey));
+
+            var responseCurr = await _fetchXMLService
+                .GetAsync<WeatherCurrent.Current>(uriCurr, headers);
+
+            if (responseCurr.IsSuccessful)
+            {
+                string message = string.Empty;
+                var data = responseCurr.Data;
+
+                var responseFore = await _fetchJsonService
+                    .GetAsync<WeatherForecast.Forecast>(uriFore, headers);
+
+                if (responseFore.IsSuccessful)
+                {
+                    var foredata = responseFore.Data;
+                    var today = foredata.Daily[0];
+
+                    message = FormatResults(data, location, today);
+                }
+                else 
+                    message = FormatResults(data, location);
+
+                return (message, null);
+            }
+
+            return (null, null);
+        }
+
+        private async Task<(string, Discord.Embed)> GetWeatherResultAsync(string input)
+        {
+            var headers = new Dictionary<string, string>
+            {
+                { "User-Agent", "DiscorIan Discord bot" }
+            };
+
+            var uri = new Uri(string.Format(_options.IanOpenWeatherMapEndpointQ,
+                HttpUtility.UrlEncode($"{input}"),
+                _options.IanOpenWeatherKey));
+
+            var response = await _fetchXMLService
+                .GetAsync<WeatherCurrent.Current>(uri, headers);
+
+            if (response.IsSuccessful)
+            {
+                var data = response.Data;
+                var locale = String.Format("{0}, {1}", data.City.Name, data.City.Country);
```

Comment:
This is fine but up above you are formatting city and state with string concatenation - it should be the same in all instances.